src/playerClass.ts

```diff
@@ -12,11 +14,34 @@ export class Player {
     }
     this._name = name;
     this._pokemons = pokemons;
+    this._activePokemon = 0;
+    this._alivePokemons = pokemons;
   }
+
   get name(): string {
     return this._name;
   }
   get pokemons(): Pokemon[] {
     return this._pokemons;
   }
+
+  set setActivePokemon(numOfActivePokemon: number) {
+    this._activePokemon = numOfActivePokemon;
+  }
+  get getActivePokemon(): number {
+    return this._activePokemon;
+  }
+
+  private isAlive() {
+    for (let i = 0; i < this._alivePokemons.length; i++) {
+      if (this._alivePokemons[i].isAlive() == false) {
+        this._alivePokemons.splice(i, 1);
+      }
+    }
+  }
+
+  get alivePokemons(): Pokemon[] {
```

Comment:
We can use filter directly on `this._pokemons`
`return this.pokemons.filter( (pokemon)=>pokemon.isAlive()`

Reply:
Okay, without create value `_alivePokemons`